### DISCUSSION TITLE
Change Version Policy for Dialogs

### DISCUSSION THREAD

**Discussion by bpauli** [adobe/aem-core-wcm-components#2047] [General]
In the Version Policy for Dialogs [1] it says that adding a new tab item is a breaking change and requires a new component version. We got the the request to add the new styles tab to all components, even the old one in #2045.

It would be nice if we can change the Version Policy and remove the requirement of a new component version when we add a new tab item to the dialog. The worst case I can think about is that customers who extended the dialog with the same dialog tab name would not get the change, or do I miss something?

https://github.com/adobe/aem-core-wcm-components/wiki/versioning-policies#dialogs

**Comment by kwin** [comment]
What about declaring a namespace for tabs which is reserved for extensions, i.e. a prefix or suffix which is never used by Core Components themselves.

**Reply by vladbailescu** [reply]
I like this idea but I would do it in an opposite manner: use an AEM-specific prefix for tabs introduced by the Core Components themselves. For example: `cq:styles`.

Renaming a tab would still be considered a breaking change as developers might have use the old name to hide it.

**Reply by bpauli** [reply]
Yes, I think for future changes and new tabs we can introduce a namespace for the Core Components themselves. But I wouldn't change the existing tabs as this would require again a new component version for all components and a lot of migration effort.

**Reply by gabrielwalt** [reply]
We could just introduce the namespace for any new tab that we're adding now, so that we can avoid creating new component version when doing so. Renaming the existing tabs isn't necessary I think. So we'd just update our component versioning rules that we reserve the `cq:` namespace for adding new tabs to existing component versions.
